packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts

```diff
@@ -610,6 +825,66 @@ export abstract class BaseService extends Resource
     this.node.defaultChild = this.resource;
   }
 
+  /**
+   *   Enable Deployment Alarms which take advantage of arbitrary alarms and configure them after service initialization
+  */
+  public enableDeploymentAlarms(alarmConfig: DeploymentAlarmConfig) {
+    // Throw an error if alarms array is empty
+    if (alarmConfig.alarms.length === 0) {
+      throw new Error('Specify at least one deployment alarm');
+    }
+    this.deploymentAlarms = {
+      alarmNames: alarmConfig.alarms.map(alarm => alarm.alarmName),
+      enable: true,
+      rollback: alarmConfig.behavior !== AlarmBehavior.FAIL_ON_ALARM,
+    };
+  }
+
+  /**
+   *   Disassociate existing deployment alarms
+  */
+  public disableDeploymentAlarms() {
```

Comment:
[API Change]: There are a few issues with this function in the current implementation.
- Can we make this private? I know you know it's not a great experience to require users to call this function when they want to update their stacks to no longer use this feature. Disabling alarms should just happen as a result of not using any of the features to configure alarms. I am going to do some testing, but it would be best to make the disabled configuration also the default configuration. This might require a feature flag if there is concern about triggering a service update for no reason for customers who upgrade their CDK and obviously didn't have any deployment alarms before.
- In the current implementation, this is always called anyway when the `deploymentAlarms` prop is not provided.
- This function will do nothing if you removed your code that configures alarms, and then also call this function. Which is what I would have assumed the customer would do if they are required to call this function in order to disable previously deployed deployment alarms.
- In any case, this needs to be well documented in the README and doc comments.

Reply:
The problem is twofold: 
1. the ECS UpdateService API is stateful. If the property is omitted from subsequent deployments, it's not removed, it's left unchanged.
2. some partitions do not support DeploymentAlarms yet; GovCloud is the major example, so we have to be careful not to specify any of the related properties (`DeployentConfiguration.Alarms: !Ref AWS::NoValue` works in pure CFN, though)

Reply:
Okay 1 is clear, and I think we can address that by making the disabled configuration the default configuration. 
(`enabled: false, alarms: []`)

I didn't know about 2, that makes it harder. Are you saying that `DeployentConfiguration.Alarms: !Ref AWS::NoValue` will work to actually remove a previously deployed deploymentAlarm? Or will CFN treat that as if the property is omitted, leaving behind any deployment alarms that were previously deployed?

Reply:
The way Copilot handles it is by specifying `NoValue` when we detect a deployment into GovCloud (handled with CFN conditions and the `AWS::Partition` pseudoparameter, but specifying `DeploymentConfiguration.enabled: false` as default otherwise. 

I believe `NoValue` makes the CFN resource handler treat the property as if it doesn't exist in the request.